src/features/extract_from_files.py

```diff
@@ -1,27 +1,39 @@
 import asyncio
 import base64
+import io
+import logging
 import os
+import typing
```

Comment:
Please do not import all of typing, just the specific objects you need - I guess that is BinaryIO?

Reply:
good point, done